nixos/modules/services/torrent/transmission.nix

```diff
@@ -21,6 +21,16 @@ let
 
   # for users in group "transmission" to have access to torrents
   fullSettings = { umask = 2; download-dir = downloadDir; incomplete-dir = incompleteDir; } // cfg.settings;
+
+  preStart = pkgs.writeScript "transmission-pre-start" ''
+    #!${pkgs.runtimeShell}
+    set -ex
+    for DIR in ${homeDir} ${settingsDir} ${fullSettings.download-dir} ${fullSettings.incomplete-dir}; do
```

Comment:
To make this future proof against overlong lines, the list of DIRs can be placed on separate lines :-)

Reply:
Good idea! Though I think I did one better (and easier actually): represent the list in nix. Reflect to shell.

Also added a test for even extra future proofing haha.